### PR TITLE
use upstream log-dump script to gather logs during e2e test

### PR DESCRIPTION
**Pull request by jaxesn**
The kubernetes/test-infra repo had a log collection script they use in the various prow jobs to test kubernetes clusters.  It handles a lot more than we really need, windows node/gke/etc, but I was thinking it would be easier to use it straight up than take on the work off modifying for our use case and maintaining.  I have left the license intact and added a short readme explaining what it is.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by bnrjee — DISMISSED** [review]
We need to remove/replace each occurrence of the word master.

File: development/kops/log-dump/log-dump.sh [hunk 1]
```diff
@@ -0,0 +1,917 @@
...
+  fi
+}
+
+function log-dump-ssh() {
+  local host="$1"
+  local cmd="$2"
+
+  if [[ "${gcloud_supported_providers}" =~ ${KUBERNETES_PROVIDER} ]]; then
+    for (( i=0; i<5; i++)); do
+      if gcloud compute ssh --ssh-flag="-o LogLevel=quiet" --ssh-flag="-o ConnectTimeout=30" --project "${PROJECT}" --zone="${ZONE}" "${host}" --command "echo test > /dev/null"; then
```
Comment (bnrjee) [inline development/kops/log-dump/log-dump.sh:265]: Do we want to use this?

File: development/kops/gather_logs.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    else
+        filter="Name=tag:k8s.io/role/node,Values=1"
+    fi
+    nodes=$(aws ec2 describe-instances \
+        --filters "Name=tag:KubernetesCluster,Values=${KOPS_CLUSTER_NAME}" "$filter" \
+        --query "Reservations[*].Instances[*].PublicDnsName" --output text)
+
+    # contaner logs are symlinks and need a slightly differnet chmod
+    # log-dump-ssh comes the upstream log-dump.sh script
+    for node in $nodes ; do log-dump-ssh $node "sudo chmod a+r /var/log/containers/*" || true ; done
```
Comment (bnrjee) [inline development/kops/gather_logs.sh:44]: We may want to store them in /logs/artifacts.

Comment (jaxesn) [reply development/kops/gather_logs.sh:44]: On prow they should be because the artifacts env is set

**Comment by jaxesn** [comment]
Fixed the non-inclusive language and added a note to the readme if we ever need to update from upstream and reproduce the changes.

File: development/kops/gather_logs.sh (hunk 2 above)
Comment (TerryHowe) [reply development/kops/gather_logs.sh:44]: Up on line 28 I assume

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: development/kops/gather_logs.sh (hunk 2 above)
Comment (jaxesn) [reply development/kops/gather_logs.sh:44]: Right, this is just the path on the instances created by kops. The log-dump script copies all of those into the artifacts path in the prow job pod.

**Pull request merged**